added checks for the 4.1 family of controls (auditd)

pkaeding opened the pull request:
I believe this faithfully performs the checks detailed in the CIS benchmark PDF for the 4.1 family of controls, relating to `auditd`.

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.1.1.rb:
```diff
@@ -23,14 +23,16 @@
+  tag nist: ["AU-12", "AU-3"]
   tag cis_level: 2
-  tag cis_controls: ["6.2", "6.3", "Rev_7"]
+  tag cis_controls: ["6.2", "6.3"]
```
can we keep the cis CDC revision as a tag perhaps?

` tag cis_cdc_version: '7'`

That way when we display the results in Heimdall we can have it auto-generate a link to the control description on the cis website.

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.1.2.rb:
```diff
@@ -26,14 +26,12 @@
+  tag nist: ["AU-12", "AU-3"]
   tag cis_level: 2
-  tag cis_controls: ["6.2", "6.3", "Rev_7"]
+  tag cis_controls: ["6.2", "6.3"]
```
same as above

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.1.3.rb:
```diff
@@ -33,14 +33,11 @@
+  tag nist: ["AU-12", "AU-3"]
   tag cis_level: 2
-  tag cis_controls: ["6.2", "6.3", "Rev_7"]
+  tag cis_controls: ["6.2", "6.3"]
```
same as above

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.1.4.rb:
```diff
@@ -41,14 +41,17 @@
+  tag cis_controls: ["6.2", "6.3"]
   tag cis_rid: "4.1.1.4"
+  grep_command = command("grep \"^\\s*linux\" /boot/grub/grub.cfg | grep -v \"audit_backlog_limit=\"").stdout.lines
+  describe grep_command do
```
do we need to add a `description` here so that the reporting tells us what this command is doing or testing?

```
grep_command = xxx
describe "The grub bootloader should have x and y" do
  subject { grep_command }
  ... 
end
```

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.10.rb:
```diff
@@ -124,14 +124,23 @@
   tag cis_rid: "4.1.10"
+
+
+  if command("uname -m").stdout.include? "x86_64"
```
Perhaps use the `os` resource here

`if os.arch.include?(/x86_64/)`

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.10.rb:
```diff
@@ -124,14 +124,23 @@
+      its('lines') { should include "-a always,exit -F arch=b64 -S creat -S open -S openat -S truncate -S ftruncate -F exit=-EPERM -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k access" }
+      its('lines') { should include "-a always,exit -F arch=b32 -S creat -S open -S openat -S truncate -S ftruncate -F exit=-EPERM -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k access" }
+    end
+  else
```
The `else` is not needed here. If we are 64 bit we are also 32 bit so we need the 32-bit tests

If we are 32-bit then we don't need the 64-bit tests

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.12.rb:
```diff
@@ -109,14 +109,18 @@
-  tag cis_controls: ["5.1", "Rev_7"]
+  tag cis_controls: ["5.1"]
   tag cis_rid: "4.1.12"
+  if command("uname -m").stdout.include? "x86_64"
```
same as above

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.11.rb:
```diff
@@ -74,14 +74,15 @@
+  tag nist: ["CM-6"]
   tag cis_level: 2
-  tag cis_controls: ["5.1", "Rev_7"]
+  tag cis_controls: ["5.1"]
```
same as above

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.13.rb:
```diff
@@ -105,14 +105,19 @@
+  tag cis_controls: ["13"]
   tag cis_rid: "4.1.13"
+
+  if command("uname -m").stdout.include? "x86_64"
```
same as above

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.13.rb:
```diff
@@ -105,14 +105,19 @@
+      its('lines') { should include "-a always,exit -F arch=b64 -S unlink -S unlinkat -S rename -S renameat -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k delete" }
+      its('lines') { should include "-a always,exit -F arch=b32 -S unlink -S unlinkat -S rename -S renameat -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k delete" }
+    end
+  else
```
same as above

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.14.rb:
```diff
@@ -41,14 +41,12 @@
+  tag nist: ["AC-2 (4)"]
   tag cis_level: 2
-  tag cis_controls: ["4.8", "Rev_7"]
+  tag cis_controls: ["4.8"]
```
same as above

aaronlippold replied:
cdc_revision

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.16.rb:
```diff
@@ -103,14 +103,24 @@
+  tag cis_controls: ["5.1"]
   tag cis_rid: "4.1.16"
+
+  if command("uname -m").stdout.include? "x86_64"
```
...

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.16.rb:
```diff
@@ -103,14 +103,24 @@
+      its('lines') { should include "-w /sbin/modprobe -p x -k modules" }
+      its('lines') { should include "-a always,exit -F arch=b64 -S init_module -S delete_module -k modules" }
+    end
+  else
```
...

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.2.1.rb:
```diff
@@ -26,14 +26,11 @@
+  tag nist: ["AU-4"]
   tag cis_level: 2
-  tag cis_controls: ["6.4", "Rev_7"]
+  tag cis_controls: ["6.4"]
```
...

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.17.rb:
```diff
@@ -30,14 +30,12 @@
+  tag nist: ["AU-12", "AU-3"]
   tag cis_level: 2
-  tag cis_controls: ["6.2", "6.3", "Rev_7"]
+  tag cis_controls: ["6.2", "6.3"]
```
...

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.2.2.rb:
```diff
@@ -25,14 +25,11 @@
+  tag nist: ["AU-4"]
   tag cis_level: 2
-  tag cis_controls: ["6.4", "Rev_7"]
+  tag cis_controls: ["6.4"]
```
...

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.2.3.rb:
```diff
@@ -39,14 +39,14 @@
+  tag nist: ["AU-4"]
   tag cis_level: 2
-  tag cis_controls: ["6.4", "Rev_7"]
+  tag cis_controls: ["6.4"]
```
...

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.3.rb:
```diff
@@ -103,14 +103,23 @@
-  tag cis_controls: ["5.5", "Rev_7"]
+  tag cis_controls: ["5.5"]
   tag cis_rid: "4.1.3"
+  if command("uname -m").stdout.include? "x86_64"
```
us the os.arch resource here

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.3.rb:
```diff
@@ -103,14 +103,23 @@
+      its('lines') { should include "-a always,exit -F arch=b32 -S clock_settime -k time-change" }
+      its('lines') { should include "-w /etc/localtime -p wa -k time-change" }
+    end
+  else
```
drop the else

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.4.rb:
```diff
@@ -60,14 +60,15 @@
+  tag nist: ["AC-2 (4)"]
   tag cis_level: 2
-  tag cis_controls: ["4.8", "Rev_7"]
+  tag cis_controls: ["4.8"]
```
...

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.5.rb:
```diff
@@ -119,14 +119,27 @@
+  tag cis_controls: ["5.5"]
   tag cis_rid: "4.1.5"
+
+  if command("uname -m").stdout.include? "x86_64"
```
same as above

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.5.rb:
```diff
@@ -119,14 +119,27 @@
+      its('lines') { should include "-w /etc/hosts -p wa -k system-locale" }
+      its('lines') { should include "-w /etc/network -p wa -k system-locale" }
+    end
+  else
```
else not needed here - these test should always render

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.6.rb:
```diff
@@ -50,14 +50,13 @@
+  tag nist: ["CM-6 (1)"]
   tag cis_level: 2
-  tag cis_controls: ["5.5", "Rev_7"]
+  tag cis_controls: ["5.5"]
```
rev tag

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.7.rb:
```diff
@@ -53,14 +53,14 @@
+  tag nist: ["AU-2", "AC-11", "AC-2(12)"]
   tag cis_level: 2
-  tag cis_controls: ["4.9", "16.11", "16.13", "Rev_7"]
+  tag cis_controls: ["4.9", "16.11", "16.13"]
```
rev_tag

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.8.rb:
```diff
@@ -57,14 +57,14 @@
+  tag nist: ["AU-2", "AC-11", "AC-2(12)"]
   tag cis_level: 2
-  tag cis_controls: ["4.9", "16.11", "16.13", "Rev_7"]
+  tag cis_controls: ["4.9", "16.11", "16.13"]
```
rev_tag

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.9.rb:
```diff
@@ -155,14 +155,25 @@
+  tag cis_controls: ["5.5"]
   tag cis_rid: "4.1.9"
+
+  if command("uname -m").stdout.include? "x86_64"
```
same as above

aaronlippold commented on controls/ubuntu-18.04-server-cis-4.1.9.rb:
```diff
@@ -155,14 +155,25 @@
+      its('lines') { should include "-a always,exit -F arch=b64 -S setxattr -S lsetxattr -S fsetxattr -S removexattr -S lremovexattr -S fremovexattr -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k perm_mod" }
+      its('lines') { should include "-a always,exit -F arch=b32 -S setxattr -S lsetxattr -S fsetxattr -S removexattr -S lremovexattr -S fremovexattr -F auid>=#{login_defs.UID_MIN} -F auid!=4294967295 -k perm_mod" }
+    end
+  else
```
these should always be run on either chipset

aaronlippold reviewed (changes requested):
Hi, it all looks good.
A few small adjustments where we can use the `os.resource` and that the 32-bit audit tests also apply to a 64-bit machine, given I can run 32-bit code on a 64-bit chip, and I think rather than removing the rev 7 data all together we should just move it into a tag. 

After that, we should be good to go and we will have a good baseline for the shape of the controls.

:)

aaronlippold commented:
Also, do you prefer I make one comment on control and you can carry the adjustment across the controls or is it easier if I make a note on each one so we don't miss one?

aaronlippold commented:
https://github.com/ExpeditionTechnology/canonical-ubuntu-18.04-lts-stig-baseline/blob/04b9f3127583e4536bb2f4c71a4b46d008bcd70b/controls/V-219278.rb is how we do the 64 vs 32 bit logic usually. 

This seems to be the cleanest way to express it

pkaeding commented:
> Also, do you prefer I make one comment on control and you can carry the adjustment across the controls or is it easier if I make a note on each one so we don't miss one?

I guess individual comments are easier-- I can use the inline comments in the file tab to confirm that I found all the issues. I'll also try to find any similar places, though.

> https://github.com/ExpeditionTechnology/canonical-ubuntu-18.04-lts-stig-baseline/blob/04b9f3127583e4536bb2f4c71a4b46d008bcd70b/controls/V-219278.rb#L68-L78
> 
> is how we do the 64 vs 32-bit logic usually.
> 
> This seems to be the cleanest way to express it I think
> 
> Put the 32 bit first then if we are 64 add those in as well ... otherwise no

Okay, that makes sense. I'll also make the changes suggested about the overlapping parts that apply for both 32bit and 64bit systems.

aaronlippold commented:
Cool thanks. Just ping me when you make the updates and I will merge once the cicd run

pkaeding commented:
> Cool thanks. Just ping me when you make the updates and I will merge once the cicd run

Okay @aaronlippold I think I've addressed all of the issues, if you want to take another look.
